ush/parsing_namelists_FV3.sh

```diff
@@ -0,0 +1,619 @@
+  knob_ugwp_lzmin    = ${knob_ugwp_lzmin:-0.75e3}
+  knob_ugwp_lzstar   = ${knob_ugwp_lzstar:-2.0e3}
+  knob_ugwp_taumin   = ${knob_ugwp_taumin:-0.25e-3}
+  knob_ugwp_tauamp   = ${knob_ugwp_tauamp:-3.0e-3}
+  knob_ugwp_lhmet    = ${knob_ugwp_lhmet:-200.0e3}
+  knob_ugwp_orosolv  = ${knob_ugwp_orosolv:-'pss-1986'}
+  $cires_ugwp_nml
+/
+EOF
+  ;;
+  *)
+    echo "FATAL: Invalid gwd_opt specified: $gwd_opt"
+    exit 1
+  ;;
+esac
+
+echo "" >> input.nml
+
+cat >> input.nml <<EOF
+&gfdl_cloud_microphysics_nml
+  sedi_transport = .true.
+  do_sedi_heat = .false.
+  rad_snow = .true.
+  rad_graupel = .true.
+  rad_rain = .true.
+  const_vi = .F.
+  const_vs = .F.
+  const_vg = .F.
+  const_vr = .F.
+  vi_max = 1.
+  vs_max = 2.
+  vg_max = 12.
+  vr_max = 12.
+  qi_lim = 1.
+  prog_ccn = .false.
+  do_qa = .true.
+  fast_sat_adj = .true.
+  tau_l2v = 225.
+  tau_v2l = 150.
+  tau_g2v = 900.
+  rthresh = 10.e-6  ! This is a key parameter for cloud water
+  dw_land  = 0.16
+  dw_ocean = 0.10
+  ql_gen = 1.0e-3
+  ql_mlt = 1.0e-3
+  qi0_crt = 8.0E-5
+  qs0_crt = 1.0e-3
+  tau_i2s = 1000.
+  c_psaci = 0.05
+  c_pgacs = 0.01
+  rh_inc = 0.30
+  rh_inr = 0.30
+  rh_ins = 0.30
+  ccn_l = 300.
+  ccn_o = 100.
+  c_paut = 0.5
+  c_cracw = 0.8
+  use_ppm = .false.
+  use_ccn = .true.
+  mono_prof = .true.
+  z_slope_liq  = .true.
+  z_slope_ice  = .true.
+  de_ice = .false.
+  fix_negative = .true.
+  icloud_f = 1
+  mp_time = 150.
+  reiflag = ${reiflag:-"2"}
+
+  $gfdl_cloud_microphysics_nml
+/
+
+&interpolator_nml
+  interp_method = 'conserve_great_circle'
+  $interpolator_nml
+/
+
+&namsfc
+  FNGLAC   = '${FNGLAC}'
+  FNMXIC   = '${FNMXIC}'
+  FNTSFC   = '${FNTSFC}'
+  FNSNOC   = '${FNSNOC}'
+  FNZORC   = '${FNZORC}'
+  FNALBC   = '${FNALBC}'
+  FNALBC2  = '${FNALBC2}'
+  FNAISC   = '${FNAISC}'
+  FNTG3C   = '${FNTG3C}'
+  FNVEGC   = '${FNVEGC}'
+  FNVETC   = '${FNVETC}'
+  FNSOTC   = '${FNSOTC}'
+  FNSMCC   = '${FNSMCC}'
+  FNMSKH   = '${FNMSKH}'
+  FNTSFA   = '${FNTSFA}'
+  FNACNA   = '${FNACNA}'
+  FNSNOA   = '${FNSNOA}'
+  FNVMNC   = '${FNVMNC}'
+  FNVMXC   = '${FNVMXC}'
+  FNSLPC   = '${FNSLPC}'
+  FNABSC   = '${FNABSC}'
+  LDEBUG = ${LDEBUG:-".false."}
+  FSMCL(2) = ${FSMCL2:-99999}
+  FSMCL(3) = ${FSMCL3:-99999}
+  FSMCL(4) = ${FSMCL4:-99999}
+  LANDICE  = ${landice:-".true."}
+  FTSFS = ${FTSFS:-90}
+  FAISL = ${FAISL:-99999}
+  FAISS = ${FAISS:-99999}
+  FSNOL = ${FSNOL:-99999}
+  FSNOS = ${FSNOS:-99999}
+  FSICL = ${FSICL:-99999}
+  FSICS = ${FSICS:-99999}
+  FTSFL = ${FTSFL:-99999}
+  FVETL = ${FVETL:-99999}
+  FSOTL = ${FSOTL:-99999}
+  FvmnL = ${FvmnL:-99999}
+  FvmxL = ${FvmxL:-99999}
+  FSLPL = ${FSLPL:-99999}
+  FABSL = ${FABSL:-99999}
+  $namsfc_nml
+/
+
+&fv_grid_nml
+  grid_file = 'INPUT/grid_spec.nc'
+  $fv_grid_nml
+/
+EOF
+
+# Add namelist for stochastic physics options
+echo "" >> input.nml
+#if [ $MEMBER -gt 0 ]; then
+if [ $do_sppt = .true. -o $do_shum = .true. -o $do_skeb = .true. ]; then
```

Comment:
I think we agreed on using `DO_SPPT=YES` for script variables logic changes.

Reply:
I really hate having variables with identical names except case that are set to different things.

Reply:
Understood